httpParser.cpp

```diff
@@ -0,0 +1,122 @@
+#include "httpParser.hpp"
+#include <sstream>
+
+httpParser::httpParser(std::map<int, Request>::iterator& pair, const WebservConfigStruct sett): req(pair), settings(sett) {
+	beheader(req->second);
+	if (req->second.HeaderBuffer.empty() == true) {
+		return;
+	}
+	if (req->second.BodyBuffer.empty() == true) {
+		return;
+	}
+	handleBody(req->second);
+}
+
+void httpParser::beheader(Request& request) {
+	size_t neck = request.RequestBuffer.find("\r\n\r\n");
+	if (neck == request.RequestBuffer.npos) {
```

Comment:
if the header is not complete this will never do anything, Because beheader will not fill the header if its not complete. header completeness is determined by a "/r/n/r/n" being present in the RequestBuffer.